cal-itp/benefits
# Feat: transit agency home dashboard

Part of #2246

This PR implements overriding the default admin site so that transit agency staff see a home dashboard upon logging in:
![image](https://github.com/user-attachments/assets/493f7830-39be-43d1-a9fd-eb3e6c21275f)

 A superuser or Cal-ITP user still sees the default Django admin site. 

Another PR will implement the top right corner "user tools" panel for transit agency staff, and another PR will implement a base `html` template that will be extended by all templates in the `in_person` app.

## Steps to test:
- Reset database to use local_fixtures (`bin/reset_db.sh`)
- Set GOOGLE_SSO_ALLOWABLE_DOMAINS, GOOGLE_SSO_STAFF_LIST, GOOGLE_SSO_SUPERUSER_LIST to appropriate values (your email and domain) in `.env`
- Go to `localhost/admin` and sign in with your email address
- Add a `CST` group
- Add the `CST` group to CST transit agency’s “staff_group” (it will ask for `audience` and `client_id`, use dummy values)
- Change your user to not superuser
- Remove `Cal-ITP` from your user’s chosen groups
- Add the `CST` group to your user’s chosen groups
- Save (will log you out)

> **_NOTE:_** If you logout and need to reach the agency home dashboard page again, you can skip the steps shown above by commenting out `add_staff_user_to_group(user, request)` in `benefits/core/admin.py`.  Then, when you log in again, your user will not be automatically added to the `Cal-ITP` group in the Google SSO prelogin hook (`pre_login_user`). If you don’t do this, you’ll need to reset the database again since you’ll never see the agency home dashboard page. The reason is that you will be part of `Cal-ITP`, but you won't be able to edit the database since you are not a superuser.

- Go to `localhost/admin`, verify that the agency home dashboard appears
